src/EnergyPlus/GlobalNames.cc

```diff
@@ -334,6 +358,41 @@ namespace GlobalNames {
 
 	}
 
+	void
+	VerifyUniqueADUName(
+		std::string const & TypeToVerify,
+		std::string const & NameToVerify,
+		bool & ErrorFound,
+		std::string const & StringToDisplay
+	)
+	{
+
+
+		ErrorFound = false;
+		int Found = 0;
+
+		if ( numAirDistUnits > 0 ) Found = FindItemInList( NameToVerify, aDUNames, &ComponentNameData::CompName, numAirDistUnits );
+
+		if ( Found != 0 ) {
+			ShowSevereError( StringToDisplay + ", duplicate name=" + NameToVerify + ", ADU Type=\"" + aDUNames( Found ).CompType + "\"" );
+			ShowContinueError( "...Current entry is Air Distribution Unit Type=\"" + TypeToVerify + "\"." );
+			ErrorFound = true;
+		} else {
+			if ( numAirDistUnits == 0 ) {
+				curMaxAirDistUnits = 4;
+				aDUNames.allocate( curMaxAirDistUnits );
+			} else if ( numAirDistUnits == curMaxAirDistUnits ) {
+				curMaxAirDistUnits += 4;
+				aDUNames.redimension( curMaxAirDistUnits );
```

Comment:
@mbadams5 If you already have an item of the array's `value_type` I suggest just using `push_back` since there is a "move" overload of it. The `emplace_back` is currently a quick placeholder and I hope to refine that and add some more vector-like API in the next couple of months. But if you have an example that should work but doesn't please send it to me. I do have a 2-iterator constructor waiting in the wings, though, which should be useful as we start using more container types.

I second the preference for `push_back` in new code (and migrating the `redimension` calls to it gradually) but a couple of things to note:
* The `curMaxAirDistUnits` then grows by 1 every time
* You can either `push_back` a default-contstructed element and then set the `CompType` and `CompName` or create such an element first and push it back. (If you could create it as a temporary inside the `push_back` that would let C++ do a move with potential performance benefits, but in practice most of our structs are still buckets of PODs so moving isn't going to make a difference yet.)
* Moving from `redimension` to `push_back` can "break" tests by changing the size of arrays that are reported in the audit files: that is OK but should be noted in the PR